contracts/SportMarkets/SportAMMRiskManager.sol

```diff
@@ -0,0 +1,206 @@
+    /// @param _totalSpent total spent on market
+    /// @param _market for which is calculation done
+    /// @return _isNotRisky true/false
+    function isTotalSpendingLessThanTotalRisk(uint _totalSpent, address _market) external view returns (bool _isNotRisky) {
+        uint capPerMarket = _calculateCapToBeUsed(_market);
+        uint riskMultiplier = _calculateRiskMultiplier(_market);
+        return _totalSpent < capPerMarket * riskMultiplier;
+    }
+
+    /* ========== INTERNALS ========== */
+
+    function _calculateRiskMultiplier(address market) internal view returns (uint toReturn) {
+        uint marketRisk = riskMultiplierPerMarket[market];
+
+        if (marketRisk == 0) {
+            (uint tag1, ) = _getTagsForMarket(market);
+            uint riskPerTag = riskMultiplierForSport[tag1];
+            marketRisk = riskPerTag > 0 ? riskPerTag : defaultRiskMultiplier;
+        }
+
+        toReturn = marketRisk;
+    }
+
+    function _calculateCapToBeUsed(address market) internal view returns (uint toReturn) {
+        toReturn = capPerMarket[market];
+        if (toReturn == 0) {
+            (uint tag1, uint tag2) = _getTagsForMarket(market);
+            uint capFirstTag = capPerSport[tag1];
+            capFirstTag = capFirstTag > 0 ? capFirstTag : defaultCapPerGame;
+            toReturn = capFirstTag;
+
+            if (tag2 > 0) {
+                uint capSecondTag = capPerSportAndChild[tag1][tag2];
+                toReturn = capSecondTag > 0 ? capSecondTag : capFirstTag / 2;
+            }
+        }
+    }
+
+    function _getTagsForMarket(address _market) internal view returns (uint tag1, uint tag2) {
+        ISportPositionalMarket sportMarket = ISportPositionalMarket(_market);
+        tag1 = sportMarket.tags(0);
+        tag2 = sportMarket.isChild() ? sportMarket.tags(1) : 0;
+    }
+
+    /* ========== CONTRACT MANAGEMENT ========== */
+
+    /// @notice Setting the Cap per spec. market
+    /// @param _markets market addresses
+    /// @param _capPerMarket The cap amount used for the specific markets
+    function setCapPerMarket(address[] memory _markets, uint _capPerMarket) external {
+        require(
+            msg.sender == owner || ISportPositionalMarketManager(manager).isWhitelistedAddress(msg.sender),
+            "Invalid sender"
+        );
+        for (uint i; i < _markets.length; i++) {
+            capPerMarket[_markets[i]] = _capPerMarket;
+            emit SetCapPerMarket(_markets[i], _capPerMarket);
+        }
+    }
+
+    /// @notice Setting the Cap per Sport ID
+    /// @param _sportID The tagID used for sport (9004)
+    /// @param _childID The tagID used for childid (10002)
+    /// @param _capPerChild The cap amount used for the sportID
+    function setCapPerSportAndChild(
+        uint _sportID,
+        uint _childID,
+        uint _capPerChild
+    ) external onlyOwner {
+        capPerSportAndChild[_sportID][_childID] = _capPerChild;
```

Comment:
Check value of all params for validity.
Ensure capPerChild is not higher than capPerParent.

Reply:
done.